annotation-server/src/fetch-dates/fetch-dates.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { FetchDate, FetchTarget } from './fetch-date.entity';
+
+@Injectable()
+export class FetchDatesService {
+    constructor(
+        @InjectRepository(FetchDate)
+        private fetchDatesRepository: Repository<FetchDate>,
+    ) {}
+
+    async get(target: FetchTarget): Promise<Date | null> {
+        const fetchDate = await this.fetchDatesRepository.findOneBy({ target });
+        return fetchDate?.date;
+    }
+
+    async set(target: FetchTarget): Promise<void> {
+        try {
+            await this.fetchDatesRepository.insert({ target });
+        } catch {
+            await this.fetchDatesRepository.update({ target }, { target });
+        }
```

Comment:
Why do you use `insert` and `update` instead of just using `save`?

Reply:
`save` tests for equality with the entity currently in the database and only updates if it's perceived as necessary (i.e. not equal). This means the date doesn't get updated since equality is always given with `target` being the only attribute.